addons/web/controllers/main.py

```diff
@@ -901,7 +901,15 @@ def do_search_read(self, model, fields=False, offset=0, limit=False, domain=None
         :rtype: list
         """
         Model = request.session.model(model)
-
+        d = [t for t in domain if not isinstance(t, basestring)]
```

Comment:
[is_leaf method for domains](https://github.com/odoo/odoo/blob/8.0/openerp/osv/expression.py#L396)